libtrellis/src/Bitstream.cpp

```diff
@@ -356,6 +361,20 @@ Chip Bitstream::deserialise_chip() {
                 rd.check_crc16();
             }
                 break;
+            case BitstreamCommand::SPI_MODE: {
+                uint8_t spi_mode;
+                rd.get_bytes(&spi_mode, 1);
+                rd.skip_bytes(2);
+
+                auto spimode = find_if(spi_modes.begin(), spi_modes.end(), [&](const pair<string, uint8_t> &fp){
+                    return fp.second == spi_mode;
+                });
+                if (spimode == spi_modes.end())
+                    throw runtime_error("bad frequency option " + spi_mode);
```

Comment:
Text should be "bad spi mode", and `+std::to_string(spi_mode) ` otherwise you'll be adding spi_mode to the chat pointer.

Reply:
Good point, fixed.